llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1126,7 +1126,8 @@ static void computeKnownBitsFromOperator(const Operator *I,
       KnownFPClass Result = computeKnownFPClass(V, fcAllFlags, Depth + 1, Q);
       FPClassTest FPClasses = Result.KnownFPClasses;
 
-      if (Result.isKnownNever(fcNormal | fcSubnormal | fcNan)) {
+      if (FPClasses != fcNone &&
```

Comment:
Can we just assume the none case is a 0?

Reply:
I think comparing with fcNone explicitly is better.

See https://github.com/llvm/llvm-project/blob/8389177710c2cb7a7e0cf5c6c9b34aabb8048f39/llvm/lib/Analysis/ValueTracking.cpp#L5007-L5008
https://github.com/llvm/llvm-project/blob/8389177710c2cb7a7e0cf5c6c9b34aabb8048f39/llvm/lib/Analysis/ValueTracking.cpp#L5174-L5178
https://github.com/llvm/llvm-project/blob/8389177710c2cb7a7e0cf5c6c9b34aabb8048f39/llvm/lib/Analysis/ValueTracking.cpp#L4199

Reply:
No, I don't mean in terms of enum. I mean treat it as-if the known bits are a literal 0

Reply:
Done.